test/widgets/inbox_test.dart

```diff
@@ -310,6 +345,30 @@ void main() {
           checkAppearsUncollapsed(tester, findSectionContent);
         });
 
+        testWidgets('collapse all-DMs section after scroll', (tester) async {
+          await setupVarious(tester);
+
+          final listFinder = find.byType(Scrollable);
+          final findSectionContent =
+              find.text(eg.otherUser.fullName).hitTestable();
+
+          // Scroll the [StickyHeaderListView] enough so that
+          // the [_AllDmsSection] shows a sticky header
+          await dragUntilInvisible(
+              tester, findSectionContent, listFinder, const Offset(0, -50));
+
+          Widget? headerRow = findAllDmsHeaderRow(tester);
+          // Check that the header is present (which in this case
+          // is a sticky one as we've scrolled enough)
+          check(headerRow).isNotNull();
+
+          await tapCollapseIcon(tester);
+          // Check that the header is still visible even after
+          // collapsing the section
+          headerRow = findAllDmsHeaderRow(tester);
+          check(headerRow).isNotNull();
```

Comment:
Rather than mutate a variable, just inline this:
```suggestion
          check(findAllDmsHeaderRow(tester)).isNotNull();
```

If the expression were longer so that this didn't make sense to inline, then it'd be better to define a new `final` variable rather than mutate the old variable. That keeps it easy to see which initializing expression each variable use is referring to.